### DISCUSSION TITLE
What issues could occur from a  default export?

### DISCUSSION THREAD

**Discussion by KiwiKilian** [callstack/react-native-builder-bob#735] [Q&A] [answered]
Taken from the [Builder Bob docs](https://callstack.github.io/react-native-builder-bob/esm#guidelines):

> Avoid using default exports in your library. Named exports are recommended. Default exports produce a CommonJS module with a default property, which will work differently than the ESM build and can cause issues.

What issues could arise from using `export default`?  This would be interesting to know, as we are [discussing to drop the default export](https://github.com/maplibre/maplibre-react-native/pull/567) on MapLibre React Native.

**Comment by satya164** [comment, marked as answer]
It's primarily an issue with dual module packages (ESM & require), by default, the template produces a library with dual module package.

You can read more at https://satya164.page/posts/publishing-dual-module-esm-libraries#default-exports

Also, the syntax to use default exports in CommonJS is not elegant as you have to do `require('./module.js').default`.

In addition, multiple exports instead of a single object in default export may help with smaller bundle sizes in bundlers that support tree-shaking (currently most web bundlers as well as expo flavored metro).

**Reply by KiwiKilian** [reply]
Thank you very much for your input!
